File: src/main/java/com/learnspark/plan/entity/Phase.java
package com.learnspark.plan.entity;

import com.learnspark.common.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

/**
 * 学习阶段实体，对应 phases 表（1 项目 : N 阶段）。
 */
@Entity
@Table(name = "phases")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Phase extends BaseEntity {

    @Id
    @Column(name = "id", updatable = false, columnDefinition = "VARCHAR(36)")
    private String id;

    @Column(name = "project_id", nullable = false, columnDefinition = "VARCHAR(36)")
    private String projectId;

    @Column(name = "name", nullable = false, length = 255)
    private String name;

    @Column(name = "objective", columnDefinition = "TEXT")
    private String objective;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @PrePersist
    void fillDefaults() {
        if (this.id == null) {
            this.id = UUID.randomUUID().toString();
        }
        if (this.sortOrder == null) {
            this.sortOrder = 0;
        }
    }
}

File: learnspark-kmp/server/src/main/kotlin/com/learnspark/server/domain/entity/Phase.kt
package com.learnspark.server.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant
import java.time.LocalDate

/**
 * 阶段实体（按文档 §11.2.3：phases 表）。
 *
 * 阶段 3.1：迁移目标。
 */
@Entity
@Table(name = "phases")
class Phase(
    @Id
    var id: String = "",

    @Column(name = "project_id", nullable = false)
    var projectId: String = "",

    @Column(nullable = false)
    var name: String = "",

    @Column(columnDefinition = "TEXT")
    var description: String? = null,

    @Column(name = "sort_order", nullable = false)
    var sortOrder: Int = 0,

    @Column(name = "start_date")
    var startDate: LocalDate? = null,

    @Column(name = "end_date")
    var endDate: LocalDate? = null,

    @Column(nullable = false, length = 20)
    var status: String = "pending",

    @Column(nullable = false)
    var version: Long = 0,

    @Column(name = "created_at", insertable = false, updatable = false)
    var createdAt: Instant? = null,

    @Column(name = "updated_at", insertable = false, updatable = false)
    var updatedAt: Instant? = null,
)
